### ISSUE TITLE
Use `tempdir` when testing static files, so temporary directories get dropped automatically

### ISSUE THREAD

**Issue by shekhirin** [paradigmxyz/reth#6824]
### Describe the feature

https://github.com/paradigmxyz/reth/blob/181883c97221d4ba1eccc0c83f6e3a717a08451d/testing/ef-tests/src/cases/blockchain_test.rs#L162-L165

Here and in other places where `create_test_static_files_dir` is used, we need to return both a guard (`TempDir`) and an actual path to the directory, so the guard gets dropped when goes out of scope and the directory gets deleted.

### Additional context

Blocked by https://github.com/paradigmxyz/reth/pull/6444

**Comment by AbnerZheng** [comment]
Can I give it a try?

**Comment by shekhirin** [comment]
@AbnerZheng hey, I assigned you, but we want to merge https://github.com/paradigmxyz/reth/pull/6444 into main first, and then accept new PRs on top of these changes.

**Comment by AbnerZheng** [comment]
NP, I can take a look first and create the PR after the PR has been merged.

**Comment by justcode740** [comment]
will take a look

**Comment by shekhirin** [comment]
hey @AbnerZheng, we just merged the PR into main! So feel free to start working on this 😄
